packages/firestore/src/local/local_documents_view.ts

```diff
@@ -56,11 +56,23 @@ export class LocalDocumentsView {
     transaction: PersistenceTransaction,
     key: DocumentKey
   ): PersistencePromise<MaybeDocument | null> {
-    return this.remoteDocumentCache
-      .getEntry(transaction, key)
-      .next(remoteDoc => {
-        return this.computeLocalDocument(transaction, key, remoteDoc);
-      });
+    return this.mutationQueue
+      .getAllMutationBatchesAffectingDocumentKey(transaction, key)
+      .next(batches => this.getDocumentInBatches(transaction, key, batches));
+  }
+
+  /** Internal version of `getDocument` that allows reusing batches. */
+  private getDocumentInBatches(
```

Comment:
No preference.   You're right that it would ideally be private... but TypeScript doesn't support "overloads" with mixed visibility. So I think `getDocumentInternal()` is good.